### ISSUE TITLE
 'overwrite': works only at root level 

### ISSUE THREAD

**Issue by ngmariusz** [dotansimha/graphql-code-generator#1334]
**Describe the bug**

Given config does not overwrite
```
{
  'require': ['ts-node/register/transpile-only'],
  'generates':
    {
      'src/modules/my-first-module/generated-models.ts':
        {
          **'overwrite': true,**
          'schema': 'src/modules/my-first-module/schema.ts',
          'plugins': ['typescript-common', 'typescript-server', 'typescript-resolvers'],
          'config': { 'noNamespace': false, 'contextType': '@graphql-modules/core#ModuleContext' },
        },
      'src/modules/my-second-module/generated-models.ts':
        {
          'overwrite': false,
          'schema': 'src/modules/my-second-module/schema.ts',
          'plugins': ['typescript-common', 'typescript-server', 'typescript-resolvers'],
          'config': { 'noNamespace': false, 'contextType': '@graphql-modules/core#ModuleContext' },
        },
    },
}
```

this overwrites each

```
{
  'require': ['ts-node/register/transpile-only'],
   **'overwrite': true,** 
  'generates':
    {
      'src/modules/my-first-module/generated-models.ts':
        {
         **'overwrite': true,** 
          'schema': 'src/modules/my-first-module/schema.ts',
          'plugins': ['typescript-common', 'typescript-server', 'typescript-resolvers'],
          'config': { 'noNamespace': false, 'contextType': '@graphql-modules/core#ModuleContext' },
        },
      'src/modules/my-second-module/generated-models.ts':
        {
         **'overwrite': false,** 
          'schema': 'src/modules/my-second-module/schema.ts',
          'plugins': ['typescript-common', 'typescript-server', 'typescript-resolvers'],
          'config': { 'noNamespace': false, 'contextType': '@graphql-modules/core#ModuleContext' },
        },
    },
}
```

**To Reproduce**

    "graphql-code-generator": "^0.16.1",
    "graphql-codegen-add": "^0.16.1",
    "graphql-codegen-time": "^0.16.1",
    "graphql-codegen-typescript-common": "0.16.1",
    "graphql-codegen-typescript-resolvers": "^0.16.1",
    "graphql-codegen-typescript-server": "^0.16.1",

     npx gql-gen

     ubuntu 16, node v8.9.4

**Comment by kamilkisiela** [comment]
#1362 should fix it

**Comment by dotansimha** [comment]
Fixed in 0.18.0 🎉
